src/Bicep.Core.Samples/Files/baselines/Resources_CRLF/main.bicep

```diff
@@ -490,3 +490,17 @@ output p4_res1childprop string = p4_child1.properties.someProp
 output p4_res1childname string = p4_child1.name
 output p4_res1childtype string = p4_child1.type
 output p4_res1childid string = p4_child1.id
+
+// parent & nested child with decorators https://github.com/Azure/bicep/issues/10970
+var dbs = ['db1', 'db2','db3']
+resource sqlServer 'Microsoft.Sql/servers@2021-11-01' = {
+  name: 'sql-server-name'
+  location: 'polandcentral'
+
+  @batchSize(1)
+  @description('Sql Databases')
+  resource sqlDatabase 'databases' = [for db in dbs: {
+    name: db
+    location: 'polandcentral'
+  }]

```

Comment:
Can we add some negative test cases to ensure other declaration types are blocked?

Reply:
Sure, will do that tomorrow.

Reply:
I added negative tests. I also needed to add visiting MissingDeclarationSyntaxes in the TypeVisitor to emit appropriate diagnostics